proxy.go

```diff
@@ -40,14 +60,16 @@ func ProxyHandler(w http.ResponseWriter, req *http.Request) {
 	// by url.Parse and the server handler
 	plain, _ := http.NewRequest(req.Method, req.URL.String(), req.Body)
 	// create a new client
-	client := NewClient(insecureSkipVerify)
-	r, err := client.Do(plain)
+
+	r, err := p.HttpClient.Do(plain)
 	if err != nil {
 		Logger.Println("Request execution error:", err)
 		w.WriteHeader(500)
 		w.Write([]byte(err.Error()))
 		return
 	}
+	// Close the connection to reuse it
```

Comment:
I had to read the Go docs on the http package to figure out what was going on here. I think the wording of this comment threw me off. The `response.Body` (not the "connection" as written) has to be closed in order for the client's next request to re-use the TCP connection.